golos.charge/golos.charge.cpp

```diff
@@ -135,7 +132,17 @@ void charge::send_charge_event(name user, const balance& state) {
     eosio::event(_self, "chargestate"_n, std::make_tuple(user, state)).send();
 }
 
-EOSIO_DISPATCH(charge, (use)(useandstore)(removestored)(setrestorer) )
+void charge::useandnotify(name user, symbol_code token_code, uint8_t charge_id, int64_t message_id, int64_t price_arg, name code) {
+    auto issuer = token::get_issuer(config::token_name, token_code);
+    require_auth(issuer);
+    auto new_val = consume_charge(issuer, user, token_code, charge_id, price_arg);
+    
+    INLINE_ACTION_SENDER(golos::publication, calcrwrdwt) (code,
```

Comment:
In this point you known only interface: account for code, action-name and arguments `account, id, new_val`. No class (golos::publication) and hardcoded action name. So you need to send action without `INLINE_ACTION_SENDER`